backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTable.java

```diff
@@ -52,11 +53,22 @@ default FinalStage supportsRowLevelSecurity() {
         }
     }
 
+    public enum SupportCase {
```

Comment:
To be fairly honnest I think the enum case name can be improved...

Also we could make it more generic...

Let me suggest:

```
public interface AlterQueryCondition extends Predicate<PostgresConfiguration> {
     AlterQueryCondition FOR_ALL = any -> true;
     AlterQueryCondition FOR_RLS_DEPLOYMENTS = conf -> conf.rowLevelSecurityEnabled();
     AlterQueryCondition FOR_NON_RLS_DEPLOYMENTS = conf -> !conf.rowLevelSecurityEnabled();
}
```

This would make it possible to add additional AlterQuery.

Reply:
```
.addAdditionalAlterQuery("CREATE INDEX mailbox_mailbox_acl_index ON " + TABLE_NAME.getName() + " USING GIN (" + MAILBOX_ACL.getName() + ")",
                PostgresTable.SupportCase.NON_RLS)
```
replacing PostgresTable.SupportCase.NON_RLS by AlterQueryCondition.FOR_NON_RLS_DEPLOYMENTS makes addAdditionalAlterQuery method more difficult to understand. Therefore, I do not want to change.

Reply:
I do actually find that the wording was not explicit enough and spent several minutes understanding what went on... Hence the suggestion. 

We could discuss the semantic of course. `NON_RLS` looks like a description of an actual state while `FOR_NON_RLS` sounds more like a condition.

As it stands I find the code not explicit enough.

Other thoughts / suggestions from team members?

Reply:
As a person who already knows the RLS context, I understand the current code.

But +1 for the `FOR`, which may help other people better understand the code.

Reply:
fixed